internal/backend/remote-state/http/backend.go

```diff
@@ -151,11 +172,33 @@ func (b *Backend) configure(ctx context.Context) error {
 
 	client := cleanhttp.DefaultPooledClient()
 
+	var tlsConfig tls.Config
 	if data.Get("skip_cert_verification").(bool) {
 		// ignores TLS verification
-		client.Transport.(*http.Transport).TLSClientConfig = &tls.Config{
-			InsecureSkipVerify: true,
+		tlsConfig.InsecureSkipVerify = true
+		client.Transport.(*http.Transport).TLSClientConfig = &tlsConfig
+	}
+	cacert := data.Get("cacert").(string)
+	if cacert != "" {
+		cacert_data, err := os.ReadFile(cacert)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", cacert, err)
+		}
+		tlsConfig.RootCAs = x509.NewCertPool()
+		if !tlsConfig.RootCAs.AppendCertsFromPEM(cacert_data) {
```

Comment:
I think all three of these arguments are being treated as PEM-encoded, right?

If so, our typical naming convention would call for them all to have `_pem` suffixes to help the reader quickly understand which format is expected, and also to potentially allow alternative forms which expect different formats later if needed. (That last part does admittedly seem relatively unlikely in this case, but it's a general convention we try to follow with strings containing serializations of data structures, such as using `_json` as a suffix for arguments which expect JSON strings.)

Combining that with my other comments above, I guess that leads to:
* `server_ca_certificate_pem`
* `client_certificate_pem`
* `client_private_key_pem`

Reply:
answered above - would rather treat cacert as client side - but add "trust" keyword that it's used to trust the other side

Reply:
I just made these changes - slight alteration… I chose 
- `client_cacert_pem`
- `client_cert_pem`
- `client_key_pem`

The nuances under the hood of `RootCA` for the client to check the server and `ClientCA` for the server to check the client are a little bass-ackwards, IMHO, but I think this naming is fine - it's _for the client_ to do something with - whether that's to "check" the server's name or to "present to" the server in TLS handshake, it's still for the client to use, phew…